src/lib/keys/KeyManagerWorker.ts

```diff
@@ -33,35 +32,33 @@ const keyManagerWorker = {
     const literals = await promisify(kbpgp.unbox)(params);
     // Get the identity that signed the data if any
     const dataSigner = literals[0].get_data_signer();
+    // get the verified message
+    const verifiedMessage = Buffer.from(literals[0].toString());
     // Retrieve the key manager associated with that data signer
-    let keyManager: any;
+    let verifiedKM: any;
     if (dataSigner) {
-      keyManager = dataSigner.get_key_manager();
+      verifiedKM = dataSigner.get_key_manager();
     }
     // If we know the pgp finger print then we say the data is verified.
     // Otherwise it is unverified.
-    if (keyManager) {
-      if (keyManager.get_pgp_fingerprint()) {
-        return true;
-      } else {
-        return false;
-      }
+    if (verifiedKM && verifiedKM.get_pgp_fingerprint() == identity.get_pgp_fingerprint()) {
+      return verifiedMessage;
     } else {
-      return false;
+      throw Error('data could not be verified');
```

Comment:
Please start to create an exception hierarchy for Polykey now.

Start with high level errors and low level errors.

We can have a `./src/errors.ts` that has all the exception classes and they extend each other like a family. That's what I usually do for smaller set of exceptions.

But if the app is larger. Then you can start having `errors.ts` for each high level model, and having those things inside `./src/<module>/errors.ts` represent the exceptions for that system and then join them together at the top level `errors.ts`.